### PR TITLE
Improved LTLf parsing

### PR DESCRIPTION
**Pull request by cipollone**
## Proposed changes

- Now, LTLf proposition symbols can only contain lower-case letters, numbers and underscores. This greately simplifies parsing and avoid mistakes with operators.
- Unary operators can be nested arbitrarely, with or without spaces. A space is still required before each proposition symbol, to avoid confusion.
- Reserved symbols, such as true, false, end, last are always reserved, regardless of capital letters.
- New tests, and adapted existing ones with lower case letters.

## Fixes

- `G X G`, `X F`, `!X` were accepted LTLf formulae, considering last symbol as propositions. Now they are refused.
- `X G a` was not accepted. Now we can also write `XG a`.
- `end` and `last` were interpreted as propositional symbols.
- Typo in `LTLfAlways.operator_symbol`.

## Types of changes

What types of changes does your code introduce to agents-tac?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -45,34 +45,34 @@
...
+    "F(G(a))",
+    "G(F(a))",
     "G(true)",
     "F(true)",
     "G(false)",
     "F(false)",
-    "last",
-    "end",
+    # "last",
+    # "end",
```
Comment (marcofavorito) [inline tests/conftest.py]: reintroduce them?

Comment (cipollone) [reply tests/conftest.py]: Yes, sure! Maybe we need a new LTLfLast class, and link to the existing LTLfEnd.

Comment (marcofavorito) [reply tests/conftest.py]: We do have `LTLfEnd` https://github.com/whitemech/flloat/blob/develop/flloat/ltlf.py#L543
But not `LTLfLast` (which is `LTLfWeakNext(LTLfFalse())`.

Ok so maybe reintroduce only `end`? `last` is for another PR?

Comment (cipollone) [reply tests/conftest.py]: I can do this before the merge.

Comment (cipollone) [reply tests/conftest.py]: > Ok so maybe reintroduce only `end`? `last` is for another PR?

I should be able to address both in the same commit.

**Comment by cipollone** [comment]
After last commit, the support for End, Last in LTLf is still missing.
Since these are parsed correctly, maybe we could merge this branch as is.
After all, they were not even supported in the develop branch, where `last` and `end` were interpreted as propositional symbols.

I think termination of traces should be addressed carefully. After talking with you, maybe we can open a new branch for that.

As an example, these assertions are all satisfied in the develop branch:
```python
a = LTLfAtomic("a")
trace = [{'a':True}]

assert LTLfWeakNext(a).truth(trace, 0)
assert LTLfWeakNext(a).truth([], 0)
assert LTLfEnd().truth([], 0)
assert LTLfEnd().truth([{'a':True}], 1)
assert not LTLfNext(LTLfEnd()).truth(trace, 0)
```
I think number 2 and 5 are surprising. Maybe it's better to address this consistently for LDLf and LTLf, instead of applying a fix.

**Comment by marcofavorito** [comment]
agreed, let's merge. Could you open an issue describing the missing features?

**Comment by marcofavorito** [comment]
Just for reference. This PR solves part of #13.

**Pull request merged**